pkg/provider/github/events.go

```diff
@@ -37,6 +43,59 @@ func (v *Provider) payloadFix(payload string) []byte {
 	return []byte(replacer.Replace(payload))
 }
 
+func (v *Provider) fetchAppToken(ctx context.Context, kube kubernetes.Interface, info *info.PacOpts, installationID int64) error {
+	// TODO: move this out of here
+	ns := os.Getenv("SYSTEM_NAMESPACE")
+	secret, err := kube.CoreV1().Secrets(ns).Get(ctx, secretName, v1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	appID := secret.Data["github-application-id"]
+	applicationID, err := strconv.ParseInt(strings.TrimSpace(string(appID)), 10, 64)
+	if err != nil {
+		return fmt.Errorf("could not parse the github application_id number from secret: %w", err)
+	}
+
+	privateKey := secret.Data["github-private-key"]
+
+	tr := http.DefaultTransport
+
+	itr, err := ghinstallation.New(tr, applicationID, installationID, privateKey)
+	if err != nil {
+		return err
+	}
+
+	// TODO: figure out this part
```

Comment:
that part is a bit complicated... there is a hack later on on token to get the unittests passing too

We maybe can do the same autodetection trick we do on gitlab here.. let's see to add that later on.